iphone/Maps/Core/Location/MWMLocationManager.mm

```diff
@@ -217,15 +217,16 @@ + (void)removeObserver:(Observer)observer
 + (void)applicationDidBecomeActive
 {
   [self start];
+  [self manager].locationManager.pausesLocationUpdatesAutomatically = YES;
 }
 
 + (void)applicationWillResignActive
 {
   BOOL const keepRunning = keepRunningInBackground();
   MWMLocationManager * manager = [self manager];
   CLLocationManager * locationManager = manager.locationManager;
-  if ([locationManager respondsToSelector:@selector(setAllowsBackgroundLocationUpdates:)])
-    [locationManager setAllowsBackgroundLocationUpdates:keepRunning];
+  [locationManager setAllowsBackgroundLocationUpdates:keepRunning];
+  locationManager.pausesLocationUpdatesAutomatically = !keepRunning;
```

Comment:
1. What are the benefits of using this logic if the background location is disabled?
2. What are the benefits of using YES value here when the app is in foreground?
3. Would it be better to set it to always NO in the initialization?

Reply:
> * What are the benefits of using this logic if the background location is disabled?

I don't understand a question. 
The logic is: 
when whe `setAllowsBackgroundLocationUpdates:true` 
the `pausesLocationUpdatesAutomatically = false` (disabled)

The app shouldn't get the location updates in BG if they aren't needed:
> When the value of this property is [true](https://developer.apple.com/documentation/swift/true?language=objc) and you start location updates while the app is in the foreground, Core Location configures the system to keep the app running to receive continuous background location updates, and arranges to show the background location indicator (blue bar or pill) if needed. **Updates continue even if the app subsequently enters the background.**

> * What are the benefits of using YES value here when the app is in foreground?

Have we ever encountered that location was paused when the app is in an Active (foreground) state? 
If not - then we should try to not change the current master's behaviour.

> * Would it be better to set it to always NO in the initialization?

I think that we should try to fix the paused **BG** location updates during the nav/track recording at first and not change the foreground behavior.

Reply:
Apple documentation and articles on the internet say that `pausesLocationUpdatesAutomatically` together with `activityType` are used exclusively to stop location updates when app is in the background. Are there any mentions of the foreground behavior influenced by these variables?

The proposal is to set `locationManager.pausesLocationUpdatesAutomatically` into `NO` during the initialization for the whole app lifecycle.

Let's assume that location updates _may_ be stopped in the foreground when the user actively uses the app. In such case, we need to also add a code to update the location button state, and to start it back again properly. If it was true, then we would already got many user reports about it.

Reply:
Ok. Lets try the `pausesLocationUpdatesAutomatically = NO` as a default option and see how it works!